src/ServiceControl.Monitoring/Infrastructure/EndpointMessageType.cs

```diff
@@ -4,7 +4,7 @@
     {
         public EndpointMessageType(string endpointName, string enclosedMessageTypes)
         {
-            var index = enclosedMessageTypes.IndexOf(';');
+            var index = (enclosedMessageTypes ?? "").IndexOf(';');
 
             var firstType = index != -1
```

Comment:
I assume it is ok to set `MessageType` to `null`?

Reply:
it is handled fine by ServicePulse, which filters out such message types

Reply:
added filter servicecontrol side (any metrics gathered against a null messagetype will be discarded anyway)